mlflow/store/artifact/s3_artifact_repo.py

```diff
@@ -110,13 +131,21 @@ def __init__(
         self._access_key_id = access_key_id
         self._secret_access_key = secret_access_key
         self._session_token = session_token
+        self.bucket, self.bucket_path = self.parse_s3_compliant_uri(self.artifact_uri)
+        temp_client = _get_s3_client(
+            access_key_id=self._access_key_id,
+            secret_access_key=self._secret_access_key,
+            session_token=self._session_token,
+        )
+        self.region_name = temp_client.get_bucket_location(Bucket=self.bucket)["LocationConstraint"]
```

Comment:
@BenWilson2 @harupy do you all have any concerns for adding this here? I'm seeing a bunch of unit test failures due to trying to run this for a fake bucket. Perhaps I can mitigate this for now by disabling `get_bucket_locaation` when `_MLFLOW_TESTING` is on?

Reply:
Never, I've decided to just fix the corresponding tests.